File: 2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findGCD = function(nums) {
    const smallest = Math.min(...nums)
    const largest = Math.max(...nums)

    return gcd(smallest, largest)
};

function gcd(a,b){
    while (b){
        [a,b] = [b, a%b]
    }
    return a
}

File: 2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.py
class Solution:
    def findGCD(self, nums: List[int]) -> int:
        smallest = nums[0]
        largest = nums[0]

        for num in nums:
            if num < smallest:
                smallest = num
            elif num > largest:
                largest = num 
                
        while largest:
            smallest, largest = largest , smallest % largest
        
        return smallest
